bitswap/client/internal/session/session.go

```diff
@@ -307,29 +308,39 @@ func (s *Session) run(ctx context.Context) {
 			switch oper.op {
 			case opReceive:
 				// Received blocks
+				trace.SpanFromContext(ctx).AddEvent("Session.ReceiveOp")
```

Comment:
Since this is in a loop, maybe better to call `span := trace.SpanFromContext(ctx)` outside loop, and then call `span.AddEvent()` inside the loop.

Reply:
Good call 🙏, previously these were all `internal.StartSpan` but didn't see the optimization here when I switched it.